ee/tabby-webserver/src/service/auth.rs

```diff
@@ -218,6 +219,55 @@ impl AuthenticationService for AuthenticationServiceImpl {
         Ok(resp)
     }
 
+    async fn request_password_reset(&self, email: &str) -> Result<()> {
+        self.get_user_by_email(email)
+            .await
+            .map_err(|_| anyhow!("The email specified is not a registered user"))?;
+        let code = self.db.create_password_reset(email.to_string()).await?;
```

Comment:
certain rate limiting logic need to be handled here.

Reply:
Any example of this rate limiting elsewhere in the code?

Reply:
you can implement it by querying the database, e.g if there's more than 3 record of pass word reset with same email in past 10 min, disallowed the request.

Reply:
That won't work, as no row will be inserted if the request fails for any reason (including user not existing), and the same user can only have one password reset code at a time.

Reply:
If we're relying on database operations to determine rate limiting, I think it defeats the purpose of rate limiting, since the rate limit checks themselves will be expensive operations.

Reply:
ok, the rate limiting is the wrong wording here. What i really mean is to have certain limitation to avoid abusing this interface to spam a certain user.

> and the same user can only have one password reset code at a time.
This is not preferred. let's use logic `if there's more than 2 record of pass word reset with same email in past 10 min, disallowed the request` to have more flexibility

Reply:
I see, I can implement something like that - prevent requesting another reset code if you have requested one in the last 5 minutes?